src/components/ThreadComponent.tsx

```diff
@@ -452,6 +524,9 @@ const ThreadComponent = ({
         <CombinedAuthenticationPage />
         <PostedSince>{parseLastModified(thread.Timestamp)}</PostedSince>
         <QuestionTitle>{thread.Title}</QuestionTitle>
+        <BookmarkButton onClick={handleBookmarkButton}>
```

Comment:
Will it be better if we add an if statement here and make the bookmark icon visible only when the user is login?